plugin/src/withAuth0.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  ConfigPlugin,
+  createRunOncePlugin,
+  withAppBuildGradle,
+  withAppDelegate,
+  withInfoPlist,
+} from '@expo/config-plugins';
+import {mergeContents} from '@expo/config-plugins/build/utils/generateCode';
+
+const pkg = require('react-native-auth0/package.json');
+
+export const addAuth0GradleValues = (
+  src: string,
+  auth0Domain?: string,
+  auth0Scheme?: string,
+): string => {
+  if (!auth0Domain) {
+    throw Error('No auth0 domain specified in expo config');
+  }
+  if (!auth0Scheme) {
+    throw Error('No auth0 scheme specified in expo config');
+  }
+  return mergeContents({
+    tag: 'react-native-auth0-manifest-placeholder',
+    src,
+    newSrc: `manifestPlaceholders = [auth0Domain: "${auth0Domain}", auth0Scheme: "${auth0Scheme}"]`,
+    anchor: /defaultConfig {/,
+    offset: 1,
+    comment: '//',
+  }).contents;
+};
+
+const withAndroidAuth0Gradle: ConfigPlugin<Auth0PluginConfig | void> = (
+  config,
+  {android, scheme, domain} = {},
+) => {
+  return withAppBuildGradle(config, config => {
+    if (config.modResults.language === 'groovy') {
+      const auth0Domain =
+        process.env.EXPO_AUTH0_DOMAIN_ANDROID ||
+        process.env.EXPO_AUTH0_DOMAIN ||
+        android?.domain ||
+        domain;
+      const auth0Scheme =
+        process.env.EXPO_AUTH0_SCHEME_ANDROID ||
+        process.env.EXPO_AUTH0_SCHEME ||
+        android?.scheme ||
+        scheme ||
+        '${applicationId}';
+
```

Comment:
Maybe add a debug log here using `debug` or otherwise which would print out what the finally resolved scheme is.